django/db/models/sql/compiler.py

```diff
@@ -2112,6 +2126,39 @@ def execute_sql(self, result_type):
                 is_empty = False
         return row_count
 
+    def execute_returning_sql(self, returning_fields):
+        """
+        Execute the specified update and return rows of the returned columns
+        associated with the specified returned_fields if the backends supports
+        it.
+        """
+        if self.query.get_related_updates():
+            raise NotImplementedError(
+                "Update returning is not implemented for queries with related updates."
+            )
```

Comment:
Surprisingly `UPDATE` for MTI (e.g. `Child.object.update(child_field="foo", parent_field="bar")`) is implemented at the compiler level. Possibly to leave the door open for it to be eventually implemented as a chain of `UPDATE` with CTEs but this will be something to consider when `update(returning)` is eventually added.

It's not an issue for `Model.save` though as each table results in a distinct `SQLUpdateCompiler.execute_returning_sql` call but I figured that adding this error would make sure that the eventual `update(returning)` logic takes this aspect into consideration.